src/posix/platform/include/openthread/openthread-system.h

```diff
@@ -219,6 +219,27 @@ const otRadioSpinelMetrics *otSysGetRadioSpinelMetrics(void);
  */
 const otRcpInterfaceMetrics *otSysGetRcpInterfaceMetrics(void);
 
+/**
+ * This function returns the ifr_flags of the infrastructure network interface.
+ *
+ * @returns The ifr_flags of infrastructure network interface.
+ *
+ */
+uint32_t otSysGetInfraNetifFlags(void);
+
+/**
+ * This functions counts Link-local Addresses, Unique Local Addresses and Global Unicast Addresses on the
+ * infrastructure network interface.
+ *
+ * @param[out] aLinkLocalAddresses      The number of Link-local Addresses.
+ * @param[out] aUniqueLocalAddresses    The number of Unique Local Addresses.
+ * @param[out] aGlobalUnicastAddresses  The number of Global Unicast Addresses.
+ *
+ */
+void otSysCountInfraNetifAddresses(uint32_t *aLinkLocalAddresses,
```

Comment:
may pass a structure parameter for easier extensions in the future? e.g. other unicast addresses or even multicast addresses, not necessary to change this general API